tests/models/api/test_llm.py

```diff
@@ -0,0 +1,157 @@
+        self.assertFalse(valid_request.beam_search)
+        self.assertEqual(len(valid_request.history), len(test_history))
+        self.assertEqual(len(valid_request.to_completion_kwargs()['messages']),
+                         2 * valid_request.max_history + 1)
+
+        # Valid explicit streaming
+        streaming_request = LLMRequest(query=test_query, history=test_history,
+                                       persona=test_persona, model=test_model,
+                                       stream=True)
+        self.assertEqual(streaming_request, valid_request)
+
+        # Valid explicit beam search
+        beam_search_request = LLMRequest(query=test_query, history=test_history,
+                                         persona=test_persona, model=test_model,
+                                         beam_search=True)
+        self.assertTrue(beam_search_request.beam_search)
+        self.assertFalse(beam_search_request.stream)
+
+        # Valid best_of, implied beam search
+        best_of_request = LLMRequest(query=test_query, history=test_history,
+                                     persona=test_persona, model=test_model,
+                                     best_of=3)
+        self.assertTrue(best_of_request.beam_search)
+        self.assertFalse(best_of_request.stream)
+
+        # Valid explicitly disable streaming and beam search
+        valid_no_stream = LLMRequest(query=test_query, history=test_history,
+                                     persona=test_persona, model=test_model,
+                                     stream=False, beam_search=False)
+        self.assertFalse(valid_no_stream.beam_search)
+        self.assertFalse(valid_no_stream.stream)
+
+        # Validate `llm` history input
+        old_history = [("user", "hello"),
+                       ("llm", "Hi, how can I help you today?"),
+                       ("user", "I am well, how about you?"),
+                       ("llm", "As a large language model, I do not feel")]
+        validated = LLMRequest(query=test_query, history=old_history,
+                               persona=test_persona, model=test_model)
+        self.assertEqual(validated.history, test_history)
+
+        # Invalid streaming with beam search
+        with self.assertRaises(ValidationError):
+            LLMRequest(query=test_query, history=test_history,
+                       persona=test_persona, model=test_model, stream=True,
+                       beam_search=True)
+        # Invalid streaming with best_of > 1
+        with self.assertRaises(ValidationError):
+            LLMRequest(query=test_query, history=test_history,
+                       persona=test_persona, model=test_model, stream=True,
+                       best_of=2)
+        # Invalid temperature with beam search
+        with self.assertRaises(ValidationError):
+            LLMRequest(query=test_query, history=test_history,
+                       persona=test_persona, model=test_model, stream=False,
```

Comment:
This should be fully valid
generation is considered `beam_search` only when `best_of`>1
If `best_of=1` it's gready search without straming and it supports temperature